code/_lighting/SimpleLight.dm

```diff
@@ -209,33 +231,35 @@
 		destroy_medium_light()
 
 /atom/proc/update_medium_light_visibility()
-	if(src.medium_lights[1].invisibility == 101) // toggled off
+	if(src.medium_lights[1].invisibility == INVIS_ALWAYS) // toggled off
 		return
-	if(!isturf(src.loc) && !isturf(src))
-		for (var/obj/overlay/simple_light/medium/light as anything in src.medium_lights)
+	if(!(isturf(src.loc) || isturf(src)))
```

Comment:
Won't both of these short circuit the same way? Not sure what the improvement is. 
Both conditions will be evaluated for the true path of the decision.